optee_lib/glue/der.c

```diff
@@ -7,44 +7,77 @@
 #include <der.h>
 #include <string.h>
 
+/*
+ * The secure element provides DER signatures while OP-TEE consumes {r,s}
+ * When verifying a signature, generating DER from {r,s} pairs is not
+ * straight forward.
+ *
+ * For simplicity we will retry signing the data when the signature provided
+ * by the secure element doesnt comply with the following restrictions.
+ *
+ * This function accepts an ECDSA DER signature from the Secure Element and
+ * returns true if OP-TEE can work with it.
+*/
+bool sss_se05x_valid_optee_ecdsa(uint8_t *der, size_t sz)
```

Comment:
Do we need to validate sz here? It's also assuming *der is valid.

Reply:
we dont usually do this (the protocol handling the APDU would have caught it) and this is supposed to be called directly after handling the APDUs...nowhere else

Reply:
Sure.